superset/docker-compose.yml

```diff
@@ -29,3 +29,14 @@ services:
       POSTGRES_PASSWORD: superset
     ports:
       - 8081:5432
+  cloud_auth_proxy:
+    image: gcr.io/cloud-sql-connectors/cloud-sql-proxy:2.13.0
+    profiles: [prod]
```

Comment:
I thought creating a separate profile for prod would be another way to avoid accidentally making changes to the production database. Think it makes sense to have to have a separate `local` profile?

Reply:
Having two profiles makes sense, I think:
- `local`: spin up the web UI + postgres; require all the auth0 stuff to be set, superset secret key, etc. 
- `prod`: just spin up cloud auth proxy and a bash shell that has all the superset dependencies - this lets you run admin commands like
  ```bash
  $ superset db upgrade
  $ superset init
  ```
  But doesn't let you do weird stuff to the prod database, like accidentally add a user through auth0.

Reply:
Sweet!

I think the superset service can just know which env it's in via the environment variables, for now.

Reply:
Is there a better way you'd like to move to in the future?

Reply:
I think we could use [env files](https://docs.docker.com/compose/environment-variables/set-environment-variables/#use-the-env_file-attribute) so we just do

```
docker compose up --profile prod --env-file prod.env # prod.env is .gitignored
docker compose up --profile local --env-file local.env # local.env is tracked in git
```

I *don't* think there's a way to set `env-file` based on `profile`, unfortunately - that seems like it'd be the nicest.

Reply:
That *seems* like it'd be a small change, actually, so if you think that sounds promising you should go for it! But I'm OK with keeping things as is for now.

Reply:
I couldn't figure out a clean way to do this but I did add some interpolation logic that requires some env vars!